File: app/editor/src/features/content/list-view/constants/logicalOperators.ts
import { LogicalOperator } from 'hooks/api-editor';
import { OptionItem } from 'tno-core';

export const logicalOperators = [
  new OptionItem('Contains', LogicalOperator.Contains),
  new OptionItem('Equals', LogicalOperator.Equals),
  new OptionItem('Not Equal', LogicalOperator.NotEqual),
  new OptionItem('Greater Than', LogicalOperator.GreaterThan),
  new OptionItem('Greater Than Or Equal', LogicalOperator.GreaterThanOrEqual),
  new OptionItem('Less Than', LogicalOperator.LessThan),
  new OptionItem('Less Than Or Equal', LogicalOperator.LessThanOrEqual),
];

File: libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/models/LogicalOperators.java
package ca.bc.gov.tno.dal.db.models;

import java.util.EnumSet;

import ca.bc.gov.tno.IEnumValue;

/**
 * Provides options for different logical operators.
 */
public enum LogicalOperators implements IEnumValue<Integer> {
  /** Contains */
  Contains(0),
  /** Equal */
  Equals(1),
  /** Not equal */
  NotEqual(2),
  /** Greater than */
  GreaterThan(3),
  /** Greater than or equal */
  GreaterThanOrEqual(4),
  /** Less than */
  LessThan(5),
  /** Less than or equal */
  LessThanOrEqual(6);

  private final int value;

  /**
   * Creates a new instance of an LogicalOperators enum value, initializes with
   * specified value.
   */
  LogicalOperators(final int newValue) {
    value = newValue;
  }

  /**
   * Get the current eum value.
   */
  public Integer getValue() {
    return value;
  }

  /**
   * Get all the logical operators enum values.
   */
  public static final EnumSet<LogicalOperators> All = EnumSet.allOf(LogicalOperators.class);
}
